site/src/App/guides/testing-guide.tsx

```diff
@@ -0,0 +1,251 @@
+        </Bullet>
+        <Bullet>Doesn&apos;t change when code is refactored</Bullet>
+        <Bullet>Encourages better API design</Bullet>
+        <Bullet>
+          Doesn&apos;t require knowledge of the implementation to be understood
+        </Bullet>
+      </BulletList>
+
+      <Text>
+        Having good tests is priceless, especially when revisiting complex
+        components that handle many edge-cases. It also makes onboarding new
+        contributors easier as they can understand the tests without
+        understanding the implementation.
+      </Text>
+
+      <Text tone="secondary">Example of testing a Menu component:</Text>
+
+      <Code playroom={false}>{`
+        test('should open menu when clicked', () => {
+          const { getByRole } = render(<MyMenu />);
+          
+          expect(getByRole('menu')).not.toBeVisible();
+  
+          const menuTrigger = getByRole('button');
+
+          menuTrigger.click();
+  
+          expect(getByRole('menu')).toBeVisible();
+        });
+      `}</Code>
+
+      <Text>
+        For more info on unit testing check out the following resources.
+      </Text>
+      <BulletList>
+        <Bullet>
+          <ExternalLink href="https://testing-library.com/docs/react-testing-library/intro">
+            React Testing Library documentation
+          </ExternalLink>
+        </Bullet>
+        <Bullet>
+          <ExternalLink href="https://kentcdodds.com/blog/testing-implementation-details">
+            Testing Implementation Details
+          </ExternalLink>
+        </Bullet>
+      </BulletList>
+
+      <Heading level="3">What about React snapshot testing?</Heading>
+
+      <Text>
+        The main benefit of snapshots tests is how quick they are to create.
+        Teams often lean on them heavily to reach an arbitrary code coverage
+        metric. However, what&apos;s often not discussed is the on-going cost
+        they add to your code base.
+      </Text>
+
+      <Text>
+        Everytime a component changes (including it&apos;s children) the
+        snapshot will change, meaning the test is broken. It&apos;s very
+        difficult to tell whether the component still does what it&apos;s
+        supposed to do just by looking at the HTML it produced. Over time, the
+        amount of snapshots you have will grow and break more frequently. In our
+        experience this causes people to grow accustomed to them changing and
+        ignore them all together. This is especially relevant when using Braid,
+        as the components will change frequently, causing many snapshots to
+        break through your app, even though it remains functional.
+      </Text>
+
+      <Text>
+        Referencing our definition of a &quot;good test&quot;, snapshots
+        don&apos;t meet any of this criteria. Instead of documenting what a
+        component should do. They document what a component does right now.
```

Comment:
These two sentences should probably be combined,

> "Instead of documenting what a component should do, they document what a component does right now."